crates/sncast/src/response/declare.rs

```diff
@@ -129,3 +135,103 @@ impl OutputLink for DeclareTransactionResponse {
 }
+
+#[derive(Clone, Serialize, Deserialize, Debug, PartialEq)]
+pub struct DeployCommandMessage {
+    accounts_file: Option<String>,
+    account: String,
+    class_hash: PaddedFelt,
+    arguments: Option<String>,
+    network_flag: String,
+}
+
+impl DeployCommandMessage {
+    pub fn new(
+        abi: &[AbiEntry],
+        response: &DeclareTransactionResponse,
+        account: &str,
+        accounts_file: &Utf8PathBuf,
+        rpc_url: Option<&str>,
+        network: Option<&Network>,
+    ) -> Result<Self, Error> {
+        let arguments = abi.iter().find_map(|entry| {
+            if let AbiEntry::Constructor(constructor) = entry {
+                let args = generate_constructor_placeholder_args(constructor.clone());
+                (!args.is_empty()).then_some(args)
+            } else {
+                None
+            }
+        });
+        let network_flag = generate_network_flag(rpc_url, network);
+        let accounts_file_str = accounts_file.to_string();
+        let accounts_file = (!accounts_file_str
+            .contains("starknet_accounts/starknet_open_zeppelin_accounts.json"))
+        .then_some(accounts_file_str);
+
+        Ok(Self {
+            account: account.to_string(),
+            accounts_file,
+            class_hash: response.class_hash,
+            arguments,
+            network_flag,
+        })
+    }
+}
+
+impl Message for DeployCommandMessage {
+    fn text(&self) -> String {
+        let mut command = String::from("sncast");
+
+        if let Some(file) = &self.accounts_file {
+            write!(command, " --accounts-file {file}").unwrap();
+        }
+
+        write!(command, " --account {}", self.account).unwrap();
+        write!(
+            command,
+            " deploy --class-hash {}",
+            self.class_hash.into_hex_string()
+        )
+        .unwrap();
+
+        if let Some(arguments) = &self.arguments {
+            write!(command, " --arguments '{arguments}'").unwrap();
+        }
+
+        write!(command, " {}", self.network_flag).unwrap();
+
+        let header = if self.arguments.is_some() {
+            "To deploy a contract of this class, replace the placeholders in `--arguments` with your actual values, then run:"
+        } else {
+            "To deploy a contract of this class, run:"
+        };
+
+        formatdoc!(
+            "
+            {header}
+            {command}
+            "
+        )
+    }
+
+    fn json(&self) -> Value {
+        // This message is only helpful in human mode, we don't need it in JSON mode.
+        Value::Null
+    }
+}
+
+fn generate_constructor_placeholder_args(constructor: AbiConstructor) -> String {
+    constructor
+        .inputs
+        .into_iter()
+        .map(|input| {
+            let input_type = input
+                .r#type
+                .split("::")
+                .last()
+                .expect("Failed to get last part of input type");
+            format!("<{} ({})>", input.name, input_type)
```

Comment:
Bit of a nitpick, but wouldn't a format like `name: Type` or `<name: Type>` be more familiar for cairo devs?

Reply:
Yeah, definitely, changed.